### PR TITLE
Add org.nickvision.money

### PR DESCRIPTION
**Pull request by nlogozzo**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

**Note:**
Money uses `--filesystem=host` for similar reasons as my other apps (Tagger and Tube Converter). C++ APIs don't work well with sandboxed permissions. A Money file should be created and opened from wherever one chooses. For example, using Money without `--filesystem=host` crashes the application when opening a database file with the error `Sqlite::Database unable to open file`. However, with `--filesystem=host`, everything works correctly.

### CODE REVIEW

**Comment by nlogozzo** [comment]
bot, build org.nickvision.money

**Comment by hfiguiere** [comment]
PR is supposed to be against the `new-pr` branch.

File: org.nickvision.money.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "43",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "org.nickvision.money",
+    "finish-args":[
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=dri",
+        "--share=ipc",
+        "--filesystem=host"
```
Comment (hfiguiere) [inline org.nickvision.money.json:12]: why do you need host access?

**Review by hfiguiere — CHANGES_REQUESTED** [review]
mssing cleanup rules whcih mean there are uneceassary files in the flatpak (headers or even static libs)

also there is an icon in this repository never referenced. It should be removed.

**Comment by nlogozzo** [comment]
> PR is supposed to be against the `new-pr` branch.

Oh, didn't know about that. I will fix it

File: org.nickvision.money.json (hunk 1 above)
Comment (nlogozzo) [reply org.nickvision.money.json:12]: Please look at the note in PR description

**Comment by nlogozzo** [comment]
> mssing cleanup rules which mean there are uneceassary files in the flatpak (headers or even static libs)
> 
> also there is an icon in this repository never referenced. It should be removed.

which icon is never referenced?

**Comment by nlogozzo** [comment]
> mssing cleanup rules which mean there are uneceassary files in the flatpak (headers or even static libs)

Oh, didn't know I needed that. Never submitted apps with cleanup rules before.

**Comment by hfiguiere** [comment]
> > mssing cleanup rules which mean there are uneceassary files in the flatpak (headers or even static libs)
> 
> Oh, didn't know I needed that. Never submitted apps with cleanup rules before.

this is not enough of a justfication to grant `host`

**Comment by nlogozzo** [comment]
> > > mssing cleanup rules which mean there are uneceassary files in the flatpak (headers or even static libs)
> > 
> > 
> > Oh, didn't know I needed that. Never submitted apps with cleanup rules before.
> 
> this is not enough of a justfication to grant `host`

```Money uses --filesystem=host for similar reasons as my other apps (Tagger and Tube Converter). C++ APIs don't work well with sandboxed permissions. A Money file should be created and opened from wherever one chooses. For example, using Money without --filesystem=host crashes the application when opening a database file with the error Sqlite::Database unable to open file. However, with --filesystem=host, everything works correctly.```

isn't enough justification? The APIs that I use can't access the files opened without it...

**Comment by hfiguiere** [comment]
> > also there is an icon in this repository never referenced. It should be removed.
> 
> which icon is never referenced?

image

**Comment by nlogozzo** [comment]
> > > also there is an icon in this repository never referenced. It should be removed.
> > 
> > 
> > which icon is never referenced?
> 
> image

That is used as the icon in the readme file...

**Comment by hfiguiere** [comment]
> isn't enough justification? The APIs that I use can't access the files opened without it...

no it is not.

**Comment by nlogozzo** [comment]
> > isn't enough justification? The APIs that I use can't access the files opened without it...
> 
> no it is not.

That's been enough justification before. Since when did all these rules change? 
And it's not my controls if the API doesn't work correctly without `--filesystem=host` . If the user doesn't support those permission, they don't need to use the app. That's the freedom of open source.

**Comment by hfiguiere** [comment]
> That is used as the icon in the readme file...
> 
> ``


No user will ever see this. It's completely unecessary.

**Comment by nlogozzo** [comment]
> 

It's nice for when the user visits the `flatpak/app-id` repo. But I guess I can remove it.

**Comment by hfiguiere** [comment]
There is less bad: `home`. Better `xdg-documents`.

Rules haven't changed. Also since you are upstream you should fix this. C++ isn't the probem.

**Comment by nlogozzo** [comment]
Would you be able to provide me with an example of cleanup rules. 
I wouldn't know what to really cleanup from the flatpak.

**Pull request closed without merging**